packages/core/database/lib/entity-manager/index.js

```diff
@@ -485,11 +504,38 @@ const createEntityManager = (db) => {
             };
           });
 
-          // if there is nothing to insert
+          // add order value when relevant
+          if (isAnyToMany(attribute)) {
+            insert.forEach((rel, idx) => {
+              rel[orderColumnName] = idx + 1;
+            });
+          }
+          // add inv_order value when relevant
+          if (isBidirectional(attribute) && isManyToAny(attribute)) {
+            const maxMap = {};
+            await Promise.all(
+              relIdsToadd.map(async (relId) => {
+                const { max } = await this.createQueryBuilder(joinTable.name)
```

Comment:
yes totally, it's in my todo. However I think I need to lock the table, the transaction would not be enough. Knex doesn't allow table locking. I might need to do raw queries but I don't like it. Any idea of how to do differently?

Reply:
We might be able to use lock for update. The max is computed based on all rows so it might prevent reading it again until the transaction has ended. Not entirely sure though

Reply:
Interesting, it's deadlocking here even though not on the same table or row.
It might be just sqlite however, I need to test with postgres and mysql.
![Capture d’écran 2022-09-19 à 15 55 33](https://user-images.githubusercontent.com/11708220/191034142-c65bbee2-ab11-4251-8051-b7d251956858.png)

Reply:
Indeed with postgres it's not the same. More than 1 connection I believe
![Capture d’écran 2022-09-19 à 16 07 12](https://user-images.githubusercontent.com/11708220/191037018-45a3c9b4-fead-4438-aeb5-677053b7b68c.png)

Reply:
`forUpdate` doesn't prevent from reading the line :/ 

![Capture d’écran 2022-09-19 à 16 20 14](https://user-images.githubusercontent.com/11708220/191040043-a6b30f2a-4554-431f-91c5-fd29f57970f2.png)

Reply:
https://stackoverflow.com/questions/52325771/manually-lock-table-in-knex

![Capture d’écran 2022-09-19 à 19 00 57](https://user-images.githubusercontent.com/11708220/191072681-27182b81-0cf0-422c-adbe-50d4e8746352.png)

Mmmh.. It won't be solved easily..